### PR TITLE
[WIP] Initial Suggestions and notebooks on decision boundary/SDT

### PR DESCRIPTION
**Pull request by pramitchoudhary**
Initial Proposal in addition to existing techniques for the paper

```
1. FI/Reason codes using Shapely Scores
2. Model-Free PDP/ICE
3. Decision Boundary (if time permits add support for selecting from principal components)
    - Validate regularization gets reflected in the boundary plots
4. Surrogates using DT (Global and Local pre and post-pruning)
5. Local Explanation linear LIME
6. Multiclass ROC plots
7. Remove reference to Eli5
8. Improve plots - if not much effort confidence bounds to all histogram plots
9. Accurate results comparison plots - review other papers and figure out 2 or illustrative plots using simulated data
10. TreeInterpreter, maybe remove it unless we are thinking of showing the visual representation?
11. Correlation feature matrix(this could be done quickly) ?

if time permits:
11. If paper looks in good condition - see if we can include some version of Rule lists(SBRL) to compliment Surrogate DT
```

Notebook related:
1. _DB (showing some early decision boundaries)
2. _DT (showing some early Srrogate DT using Sklearn DT(pre and post-pruning)

For comparing FI, we could also compare the default FI given by the model(if the model supports it) and compare if there is a difference between Shapely computed score n Model specific FI(explicitly validating that the results could be different)

### CODE REVIEW

**Comment by navdeep-G** [comment]
Are the changes to the xgb_credit_multinomial_shap_pdp_ice.ipynb needed? If not, please reset back to original to avoid the above conflict.

**Comment by pramitchoudhary** [comment]
If we need another use-case Lending Club dataset could be another one.
Here is an example notebook for that
https://github.com/datascienceinc/Skater/tree/master/examples/credit_analysis

**Comment by pramitchoudhary** [comment]
@navdeep-G 
there should be 4 notebooks(_ignore the original xgb_simulated_multinomial_shap_pdp_ice.ipynb_). When rebasing, to avoid any unwanted to merge conflict, if you could rename xgb_simulated_multinomial_shap_pdp_ice.ipynb to xgb_simulated_multinomial_shap_pdp_ice_bkup.ipynb that would be great help.

Let me know if something appears to be ambiguous. I need to still improve the notebooks with more comments and context. We will do it as we write more paper writing.

**Comment by navdeep-G** [comment]
I would close this PR, open a new one with your new notebooks, and reopen another PR. There should be no need for the master branch to change.
